poly-commitment/src/pairing_proof.rs

```diff
@@ -91,23 +91,22 @@ impl<
 {
     type SRS = PairingSRS<Pair>;
 
-    fn open<EFqSponge, RNG, D: EvaluationDomain<<G as AffineCurve>::ScalarField>>(
+    fn open<EFqSponge, RNG, D: EvaluationDomain<F>>(
         srs: &Self::SRS,
         _group_map: &<G as CommitmentCurve>::Map,
         plnms: &[(
-            DensePolynomialOrEvaluations<<G as AffineCurve>::ScalarField, D>,
+            DensePolynomialOrEvaluations<F, D>,
             Option<usize>,
-            PolyComm<<G as AffineCurve>::ScalarField>,
+            PolyComm<F>,
         )], // vector of polynomial with optional degree bound and commitment randomness
-        elm: &[<G as AffineCurve>::ScalarField], // vector of evaluation points
-        polyscale: <G as AffineCurve>::ScalarField, // scaling factor for polynoms
-        _evalscale: <G as AffineCurve>::ScalarField, // scaling factor for evaluation point powers
-        _sponge: EFqSponge,                      // sponge
+        elm: &[F],          // vector of evaluation points
+        polyscale: F,       // scaling factor for polynoms
+        _evalscale: F,      // scaling factor for evaluation point powers
+        _sponge: EFqSponge, // sponge
         _rng: &mut RNG,
     ) -> Self
     where
-        EFqSponge:
-            Clone + FqSponge<<G as AffineCurve>::BaseField, G, <G as AffineCurve>::ScalarField>,
+        EFqSponge: Clone + FqSponge<<G as AffineCurve>::BaseField, G, F>,
```

Comment:
Unless you disagree, can you also please add the constraint `F: <G as AffineCurve>::ScalarField` that was removed? I feel like it was not on purpose if you wanted to simplify things... `F` is definitely better for readability, but unconstraining it to be any field is more confusing since then looking at type definition would make me think "this `F` could be anything".

Reply:
Ah I see! Makes sense :+1: